src/reanimated2/animation/clamp.ts

```diff
@@ -0,0 +1,92 @@
+'use strict';
+import { defineAnimation, getReduceMotionForAnimation } from './util';
+import type {
+  Animation,
+  Timestamp,
+  AnimatableValue,
+  AnimationObject,
+  ReduceMotion,
+} from '../commonTypes';
+import type { ClampAnimation } from './commonTypes';
+
+type withClampType = <T extends AnimatableValue>(
+  clamp: { min: number; max: number },
+  delayedAnimation: T,
+  reduceMotion?: ReduceMotion
+) => T;
+
+// TODO This feature is not documented yet
```

Comment:
I think that documentation should be done in separate PR

Reply:
Could be, but maybe we can try in this one?

Reply:
Could we add examples then?
In our examples in docs we use reanimated from node_modules. We can't add example using a function that hasn't been merged yet. I suppose that splitting this in two PRs is a better solution.